### PR TITLE
adding filters (for lambda policies) to resources missing the property

### PR DESCRIPTION
**Pull request by alfredgamulo**

### CODE REVIEW

**Comment by alfredgamulo** [comment]
ml and iot are awkward so still leaving this test commented out:
https://github.com/capitalone/cloud-custodian/blob/master/tests/test_policy.py#L90

File: c7n/resources/gamelift.py [hunk 1]
```diff
@@ -41,3 +42,5 @@ class resource_type(object):
         dimension = None
         batch_detail_spec = (
             "describe_fleet_attributes", "FleetIds", None, "FleetAttributes")
+        filter_name = 'BuildId'
```
Comment (kapilt) [inline c7n/resources/gamelift.py]: that doesn't seem quite right. buildid is a parent for potentially a set of fleets (ie. its a parent).

File: c7n/resources/cloudfront.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ class resource_type(object):
         date = 'LastModifiedTime'
         dimension = "DistributionId"
         universal_taggable = True
+        filter_name = None
```
Comment (kapilt) [inline c7n/resources/cloudfront.py]: hmmm.. i'm wondering if we need a differential notion besides enumeration filters, getting all fronts on lambda to pick up one seems odd, if we have the ids technically we could just pass directly into detail spec to pick up the resources.

**Comment by kapilt** [comment]
rather than submitting several two line pull requests that are all related to the same change it would be better to keep then grouped re #1589 and #1590 given their very small size either apart or in aggregate.

File: c7n/resources/gamelift.py (hunk 1 above)
Comment (alfredgamulo) [reply c7n/resources/gamelift.py]: From the boto3 docs the list_fleets does use buildID as a filter:
```
response = client.list_fleets(
    BuildId='string',
    Limit=123,
    NextToken='string'
)
```

and the cloudtrail includes the info:
```
"eventSource": "gamelift.amazonaws.com",
    "eventName": "CreateFleet",
    "awsRegion": "us-east-1",
    "sourceIPAddress": "XXX.XXX.XXX.XXX",
    "userAgent": "signin.amazonaws.com",
    "requestParameters": {
        "buildId": "build-c6e5be7e-c55a-41c1-952b-eba575fab016",
        "eC2InboundPermissions": [
            {
                "ipRange": "0.0.0.0/0",
                "fromPort": 33435,
                "protocol": "UDP",
                "toPort": 33435
            }
        ],
        "eC2InstanceType": "c4.large",
        "serverLaunchPath": "C: \\game\\Bin64.Release.Dedicated\\MultiplayerProjectLauncher_Server.exe",
        "metricGroups": [
            "default"
        ],
        "description": "ThisFleetwascreatedfromtheSamplepage.",
        "serverLaunchParameters": "+sv_port33435+gamelift_start_server",
        "name": "GameLiftSampleFleet: wowcraft"
    },
    "responseElements": {
        "fleetAttributes": {
            "fleetId": "fleet-771b29e6-c085-4c3f-b049-48647931f7d6",
            "serverLaunchPath": "C: \\game\\Bin64.Release.Dedicated\\MultiplayerProjectLauncher_Server.exe",
            "newGameSessionProtectionPolicy": "NoProtection",
            "status": "NEW",
            "creationTime": "Sep22,
20171: 12: 42PM",
            "buildId": "build-c6e5be7e-c55a-41c1-952b-eba575fab016",
            "metricGroups": [
                "default"
            ],
            "operatingSystem": "WINDOWS_2012",
            "description": "ThisFleetwascreatedfromtheSamplepage.",
            "serverLaunchParameters": "+sv_port33435+gamelift_start_server",
            "name": "GameLiftSampleFleet: wowcraft",
            "fleetArn": "arn: aws: gamelift: us-east-1: : fleet/fleet-771b29e6-c085-4c3f-b049-48647931f7d6"
        }
    }
```

**Comment by alfredgamulo** [comment]
merged the smaller PRs into this one.
thought process was that I'd like those small ones merged in while this PR might stay open for longer due to ml and iot still being outstanding and further discussion on refactors using detail_spec where available.

File: c7n/resources/gamelift.py (hunk 1 above)
Comment (kapilt) [reply c7n/resources/gamelift.py]: build id is a build number, if you query on build id, what will you get.. all the fleets that were created with that build id, what was the event for ... one fleet.

another example in the same context, say you wanted to do a policy on create of an ecs service, if you did the same you'd be passing in cluster id, and then evaluating all the services in a cluster including those that already exist, as opposed to the one that was actually evented.

**Comment by kapilt** [comment]
add an issue for the consideration of detail_spec it needs some design discussion.

**Pull request merged**